agent/agent.go

```diff
@@ -158,9 +158,9 @@ func (a *Agent) run(ctx context.Context) {
 		select {
 		case operation := <-sessionq:
 			operation.response <- operation.fn(session)
-		case msg := <-session.tasks:
-			if err := a.worker.Assign(ctx, msg.Tasks); err != nil {
-				log.G(ctx).WithError(err).Error("task assignment failed")
+		case msg := <-session.assignments:
+			if err := a.worker.Update(ctx, msg.UpdateTasks, msg.RemoveTasks, msg.Type); err != nil {
```

Comment:
These leaks the assignment message type into the worker model.

This should be something like the following:

```
switch msg.Type {
case COMPLETE:
   a.worker.Assign(...)
case INCREMENTAL:
   a.worker.Update(...)
}
```